RELEASE.md

```diff
@@ -2,19 +2,25 @@
 
 The Kubebuilder Project is released on an as-needed basis. The process is as follows:
 
-1. An issue is proposing a new release with a changelog since the last release
+1. An issue is proposing a new release with a changelog since the last release. You will need to use the [kubebuilder-release-tools][kubebuilder-release-tools] to generate the notes. See [here][release-notes-generation]
 1. All [OWNERS](OWNERS) must LGTM this release
-1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`
-1. A PR needs to be created to merge `master` branch into `book-v2` to pick up the new docs. 
+1. An OWNER runs `git tag -s $VERSION` and inserts the changelog and pushes the tag with `git push $VERSION`. Note that after you push the tag the `ci` will automatically add the release notes and the assets. 
+1. A PR needs to be created to merge `release-3` branch into `book-v3` to pick up the new docs. 
```

Comment:
Using generic versions should be prefered so that we do not need to update instructions every release.
```suggestion
1. A PR needs to be created to merge `release-X` branch into `book-vX` to pick up the new docs. 
```